### PR TITLE
[WIP] Docker: Fix worker Dockerfile to be build by OBS

### PR DESCRIPTION
**Pull request by ilausuch**
Problem: OBS cannot deal with the current repositories (OBS)
Solution: Change the repositories to be downloaded from
http://download.opensuse.org

Problem: The default OS Leap version for containers in OBS is 15.1
Solution: Fix the Dockerfile to use Leap 15.1 instead

Problem: OBS doesn't have a correct tag when build the images
Solution: Provide a tag

https://progress.opensuse.org/issues/43715

### CODE REVIEW

File: docker/worker/Dockerfile_x86 [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!BuildTag: openqa_worker_x86
```
Comment (kalikiana) [inline docker/worker/Dockerfile_x86:1]: Why is this needed? Do you have a ticket or documentation we can link here? Or a quote from a friendly OBS expert perhaps?

File: docker/worker/Dockerfile_arm [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!BuildTag: openqa_worker_arm
```
Comment (kalikiana) [inline docker/worker/Dockerfile_arm:1]: How does this relate to `docker/worker/Dockerfile`? Should that be deleted or replaced?

File: docker/worker/Dockerfile_x86 [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+FROM opensuse/leap:15.1
+LABEL maintainer Jan Sedlak <jsedlak@redhat.com>, Josef Skladanka <jskladan@redhat.com>, wnereiz <wnereiz@eienteiland.org>, Sergio Lindo Mansilla <slindomansilla@suse.com>
+LABEL version="0.3"
+
+RUN zypper ar -f -G http://download.opensuse.org/repositories/devel:/openQA:/Leap:/15.1/openSUSE_Leap_15.1 openQA
```
Comment (kalikiana) [inline docker/worker/Dockerfile_x86:6]: Is there a ticket explaining why `http://` must be used? Is this x86_64-specific?

File: docker/worker/Dockerfile_x86 [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!BuildTag: openqa_worker_x86
+FROM opensuse/leap:15.1
```
Comment (kalikiana) [inline docker/worker/Dockerfile_x86:2]: Why 15.1 on x86_64 and not 15.2? 🤔

**Review by kalikiana — COMMENTED** [review]
I realize this is a draft, but I added the questions that came up earlier while investigating the problems with the build, and I think it would be very useful to understand why the respective things are needed.

File: docker/worker/Dockerfile_x86 (hunk 1 above)
Comment (perlpunk) [reply docker/worker/Dockerfile_x86:1]: That's the tag the container will get.
We have this here too: https://github.com/os-autoinst/os-autoinst/blob/master/docker/travis_test/Dockerfile#L1

**Review by okurz — CHANGES_REQUESTED** [review]
The repo inconsistency shows the problem of maintainability already. We should find a better way the prevent the duplication. Maybe extract a shell script. Maybe even build container images in OBS not using Dockerfile but kiwi profiles if that's easier

File: docker/worker/Dockerfile_x86 (hunk 4 above)
Comment (Martchus) [reply docker/worker/Dockerfile_x86:2]: I've been asking myself the same question.

**Comment by ilausuch** [comment]
Closing this PR. The good one is https://github.com/os-autoinst/openQA/pull/3533
